packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWRelatedCommunityCard/CWRelatedCommunityCard.tsx

```diff
@@ -1,112 +1,200 @@
+import { notifyError } from 'client/scripts/controllers/app/notifications';
 import type ChainInfo from 'client/scripts/models/ChainInfo';
+import { useManageCommunityStakeModalStore } from 'client/scripts/state/ui/modals';
+import { CommunityData } from 'client/scripts/views/pages/DirectoryPage/DirectoryPageContent';
 import { isCommandClick, pluralizeWithoutNumberPrefix } from 'helpers';
 import { useBrowserAnalyticsTrack } from 'hooks/useBrowserAnalyticsTrack';
 import { navigateToCommunity, useCommonNavigate } from 'navigation/helpers';
-import React, { useCallback } from 'react';
+import React, { useCallback, useState } from 'react';
+import app from 'state';
 import {
   MixpanelClickthroughEvent,
   MixpanelClickthroughPayload,
 } from '../../../../../../../shared/analytics/types';
+import { useCommunityStake } from '../../../CommunityStake';
+import useJoinCommunity from '../../../SublayoutHeader/useJoinCommunity';
 import { CWCommunityAvatar } from '../../cw_community_avatar';
 import { CWIcon } from '../../cw_icons/cw_icon';
 import { CWText } from '../../cw_text';
 import { ComponentType } from '../../types';
+import { CWButton } from '../cw_button';
 import './CWRelatedCommunityCard.scss';
 import { addPeriodToText } from './utils';
 
 type CWRelatedCommunityCardProps = {
-  id: string;
-  communityName: string;
-  communityIconUrl: string;
-  communityDescription: string;
-  memberCount: string;
-  threadCount: string;
-  actions?: JSX.Element;
+  community: ChainInfo | CommunityData;
+  memberCount: string | number;
+  threadCount: string | number;
+  stakeChange?: number;
+  setSelectedCommunity?: React.Dispatch<
+    React.SetStateAction<ChainInfo | CommunityData>
+  >;
 };
 
 export const CWRelatedCommunityCard = ({
-  id,
-  communityName,
-  communityIconUrl,
-  communityDescription,
+  community,
   memberCount,
   threadCount,
-  actions,
+  stakeChange,
+  setSelectedCommunity,
 }: CWRelatedCommunityCardProps) => {
   const navigate = useCommonNavigate();
-  const communityAvatar = {
-    iconUrl: communityIconUrl,
-    name: communityName,
-  } as ChainInfo;
+  const { stakeEnabled, stakeValue } = useCommunityStake({
+    community: community,
+  });
+
+  const { handleJoinCommunity, JoinCommunityModals } = useJoinCommunity({
+    communityToJoin: community.id,
+  });
+  const [hasJoinedCommunity, setHasJoinedCommunity] = useState(false);
+  const isCommunityMember =
+    app.roles.getAllRolesInCommunity({ community: community.id }).length > 0;
```

Comment:
can we modify `Permissions.isCommunityAdmin()` (from `client/scripts/utils/Permissions.ts`) for this check, this would help in other areas